istioctl/cmd/webhook_test.go

```diff
@@ -0,0 +1,304 @@
+whBUEPSkGBFe/eMPSFWBW27a0nw0cK2s/5yuFhTRtcUrZ9+ojJg4IS3oSm2UZ6UJ
+DuNI3qwB6OlPQOcWX8uEp4eAaolD1lIbLRQYvxYrBqnyCZBLE+MJgA1/VB3dAECB
+TxPtAqcwLFcvsM5ABys8yK8FrqRn5Bx54NiztgG+yU30W33xjdqzmEmuIIk4JjPU
+ZQRsug7XClDvQKM6lbYcYS1td2zT08hdgURFXJ9VR64ALFp00/bvglpryu8FmQID
+AQABo1MwUTAMBgNVHRMBAf8EAjAAMEEGA1UdEQQ6MDiCHHByb3RvbXV0YXRlLmlz
+dGlvLXN5c3RlbS5zdmOCGHByb3RvbXV0YXRlLmlzdGlvLXN5c3RlbTANBgkqhkiG
+9w0BAQsFAAOCAQEAhcVEZSuNMqMUJrWVb3b+6pmw9o1f7j6a51KWxOiIl6YuTYFS
+WaR0lHSW8wLesjsjm1awWO/F3QRuYWbalANy7434GMAGF53u/uc+Z8aE3EItER9o
+SpAJos6OfJqyok7JXDdOYRDD5/hBerj68R9llWzNJd27/1jZ0NF2sIE1W4QFddy/
++8YA4+IqwkWB5/LbeRznl3EjFZDpCEJk0gg5XwAR5eIEy4QU8GueTwrDkssFdBGq
+0naco7/Es7CWQscYdKHAgYgk0UAyu8sGV235Uw3hlOrbZ/kqvyUmsSujgT8irmDV
+e+5z6MTAO6ktvHdQlSuH6ARn47bJrZOlkttAhg==
+-----END CERTIFICATE-----
+`
+)
+
+var (
+	fakeCACert = []byte("fake-CA-cert")
+)
+
+func TestCheckCertificate(t *testing.T) {
+	testCases := map[string]struct {
+		cert       string
+		shouldFail bool
+	}{
+		"cert valid": {
+			cert:       exampleCACert,
+			shouldFail: false,
+		},
+		"cert invalid": {
+			cert:       "invalid-cert",
+			shouldFail: true,
+		},
+	}
+
+	for _, tc := range testCases {
+		err := checkCertificate([]byte(tc.cert))
+		if tc.shouldFail {
+			if err == nil {
+				t.Errorf("should have failed")
+			}
+		} else if err != nil {
+			t.Errorf("should not fail, but err: %v", err)
+		}
+	}
+}
+
+func TestVerifyCertChain(t *testing.T) {
+	testCases := map[string]struct {
+		cert       string
+		caCert     string
+		shouldFail bool
+	}{
+		"cert chain valid": {
+			cert:       exampleIssuedCert,
+			caCert:     exampleCACert,
+			shouldFail: false,
+		},
+		"cert chain invalid": {
+			cert:       exampleCACert,
+			caCert:     exampleIssuedCert,
+			shouldFail: true,
+		},
+	}
+
+	for _, tc := range testCases {
+		err := veriyCertChain([]byte(tc.cert), []byte(tc.caCert))
+		if tc.shouldFail {
+			if err == nil {
+				t.Errorf("should have failed")
+			}
+		} else if err != nil {
+			t.Errorf("should not fail, but err: %v", err)
+		}
+	}
+}
+
+func TestBuildValidatingWebhookConfig(t *testing.T) {
+	configFile := "./testdata/webhook/example-validating-webhook-config.yaml"
+	configName := "proto-validate"
+	webhookConfig, err := buildValidatingWebhookConfig(fakeCACert, configFile, configName)
+	if err != nil {
+		t.Fatalf("err building validating webhook config %v: %v", configName, err)
+	}
+	if webhookConfig.Name != configName {
+		t.Fatalf("webhookConfig.Name (%v) is different from %v", webhookConfig.Name, configName)
+	}
+	for i := range webhookConfig.Webhooks {
+		if !bytes.Equal(webhookConfig.Webhooks[i].ClientConfig.CABundle, fakeCACert) {
+			t.Fatalf("webhookConfig CA bundle(%v) is different from %v",
+				webhookConfig.Webhooks[i].ClientConfig.CABundle, fakeCACert)
+		}
+	}
+}
+
+func TestDisableCliOptionsValidation(t *testing.T) {
+	testCases := map[string]struct {
+		opt        disableCliOptions
+		shouldFail bool
+	}{
+		"valid option 1": {
```

Comment:
More descriptive names would help. You could probably just print the entire set of options in this case.

Reply:
Done: print the entire set when error occurs.